### ISSUE TITLE
Organised Crime Life Module

### ISSUE THREAD

**Issue by SeanPGorman** [chummer5a/chummer5a#932]
Please fill out the below as accurately as possible. More detail makes it easier for us to find the problems and fix them.
### Chummer Version

181
### Expected behaviour

Uncertain.
### Actual behaviour

Selecting the life module Organised crime causes a handled error reading: "No selectable contacts found. Please create a contact and try again."
Pressing Ok at this point creates a Positive Quality that has no content and a prompt for the Value of SINner (Criminal) Enter something in here and then press ok.
Click on the Null positive quality.

Retried but added a Contact prior to using the Life Module facility but the performance was the same.
### Steps to reproduce the behaviour

See above
### Link to affected character file

[Test Organised Crime.txt](https://github.com/chummer5a/chummer5a/files/317526/Test.Organised.Crime.txt)

**Comment by SeanPGorman** [comment]
It might be the GUID on the Made Man quality:

		<quality>
			<guid>00000000-0000-0000-0000-000000000000</guid>
			<name>Made Man</name>
			<extra />
			<bp>5</bp>
			<implemented>True</implemented>
			<contributetolimit>True</contributetolimit>
			<metagenetic />
			<print>True</print>
			<qualitytype>Positive</qualitytype>
			<qualitysource>Selected</qualitysource>
			<source>RF</source>
			<page>148</page>
			<bonus />
			<notes />
		</quality>

**Comment by SeanPGorman** [comment]
Looked through the Qualities xml and saw what might be the issue:
This looks like the first **mademan /** attribute is misplaced perhaps? Just making an assumption by looking at the shape of the other qualities.

`		<quality>
			<name>Made Man</name>
			<karma>5</karma>
			<category>Positive</category>
			<bonus>
				<mademan />
				<selectcontact>
					<type>group</type>
					<mademan />
				</selectcontact>
			</bonus>
			<source>RF</source>
			<page>148</page>
		</quality>`

**Comment by joha4270** [comment]
It is not, they both serve different purposes. If they should be merged
into one (everything in the first _could_ be done in the second) is another
question, but it is working and i don't see any reason to fix what ain't
broken.

**Comment by joha4270** [comment]
Well, certain definition of working, but i'm 80% sure the issue is located somewhere else.

**Comment by SeanPGorman** [comment]
oh. No I see: "there is an issue, but this isn't it."

Thanks Johannes

**Comment by SeanPGorman** [comment]
When I try to add the Made Man positive quality to a character with either Priorities or Life Modules then I get the error. I have tried many types of contact and none of them seem to matter.

This issue is definitely associated to the Made Man quality and I think the issue causes the empty display name of the quality while the GUID of the quality is all 0s (see image above).

**Comment by joha4270** [comment]
Have you tried to mark one of your contracts as a group before you added the Organised crime?

**Comment by chummer5a** [comment]
Okay, so the problem is that by the time we hit the selectcontact node for
the made man quality, we've already started generating other segments of
the life module, but failing out of the improvement creation leaves
everything half-done.

There are a couple of ways we can fix this; first would be code that
prevents you from taking the life module without having a contact that
qualifies. Second would be to build in functionality to generate the
appropriate contact if one isn't found. Both have drawbacks and caveats,
but I'd suggest the former would have less impact.

**Comment by SeanPGorman** [comment]
Ok - so this works if the contact is listed as a member of a group. I made an assumption that the listing for Mafia Don would imply membership of a group and didn't understand the relevance of the group box.

I have tested this with both Life Modules and Priority and the quality works fine with the above restriction. Users must have a Contact with the Group checkbox set to select Organised Crime to prevent a fatal crash (which occurs when the user clicks on the empty space in the Qualities list.)

Is there a way to call a dialog to create a contact when a MadeMan call is made? To make all of the other elements work in series so that if the Cancel is clicked then the whole method fails?

When the MadeMan quality is selected by itself then there is no crash but there is also no validation message to show that the process is failing elegantly.
When the MadeMan quality is added by selecting the Organised Crime Life Module the Karma is taken off and the empty quality row is added.

My choice, if it matters and there is no way to add a new 'AddContact Form', then I'd suggest the second option: Add the Contact by rote. The only issue here is that the Quality **Made Man (John Smith)** does not update should I change the name of the Contact. If that cannot be modified then Option 1 makes sense.
